database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -306,7 +306,7 @@ protected void writeToDB(PreparedStatement stmt, @Nullable Schema.Field field, i
           break;
         case TIMESTAMP_MILLIS:
         case TIMESTAMP_MICROS:
-          stmt.setTimestamp(sqlIndex, Timestamp.from(record.getTimestamp(fieldName).toInstant()));
+          stmt.setTimestamp(sqlIndex, Timestamp.valueOf(record.getTimestamp(fieldName).toLocalDateTime()));
```

Comment:
why is this change needed?

Reply:
During the read operation in the database source plugin we convert from local to Gregorian, which is not getting converted back to local time during write. This issue causes the offset on DB-DB pipelines. Related JIRA : https://cdap.atlassian.net/browse/PLUGIN-1518, https://cdap.atlassian.net/browse/PLUGIN-1526